tests/app/notifications/test_process_notification.py

```diff
@@ -529,3 +529,89 @@ def test_persist_notification_for_international_letter(sample_letter_template, p
     persisted_notification = Notification.query.get(notification.id)
     assert persisted_notification.postage == postage
     assert persisted_notification.international
+
+
+@pytest.mark.parametrize(
+    "unsubscribe_link",
+    ["https://please-unsubscribe-me.com/unsubscribe", None],
+)
+def test_persist_notification_when_template_has_unsubscribe_link_is_false(
+    unsubscribe_link, mocker, sample_job, sample_api_key
+):
```

Comment:
maybe we could have a short docstring for this test, too, like we do for one below - to explain what w're testing? For example:
```
If template_has_unsubscribe_link is False, we either use the unsubscribe_link provided by the user, or set it to None.
```